molgenis-app/src/main/java/org/molgenis/app/promise/mapper/ParelMapper.java

```diff
@@ -0,0 +1,309 @@
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.MATERIALS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.NAME;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.NUMBER_OF_DONORS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.OMICS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.PRINCIPAL_INVESTIGATORS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.PUBLICATIONS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_AGE_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_COLLECTION_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_COUNTRIES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_DATA_CATEGORY_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_DISEASE_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_GENDER_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_JURISTIC_PERSONS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_MATERIAL_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_OMICS_DATA_TYPES;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.REF_PERSONS;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.SAMPLE_COLLECTIONS_ENTITY;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.SEX;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.TYPE;
+import static org.molgenis.app.promise.model.BbmriNlCheatSheet.WEBSITE;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.molgenis.app.promise.client.PromiseDataParser;
+import org.molgenis.app.promise.mapper.MappingReport.Status;
+import org.molgenis.app.promise.model.BbmriNlCheatSheet;
+import org.molgenis.app.promise.model.PromiseMappingProjectMetaData;
+import org.molgenis.data.DataService;
+import org.molgenis.data.Entity;
+import org.molgenis.data.EntityMetaData;
+import org.molgenis.data.support.MapEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+
+@Component
+public class ParelMapper implements PromiseMapper, ApplicationListener<ContextRefreshedEvent>
+{
+	private final String ID = "PAREL";
+
+	private PromiseMapperFactory promiseMapperFactory;
+	private PromiseDataParser promiseDataParser;
+	private DataService dataService;
+
+	private static final Logger LOG = LoggerFactory.getLogger(ParelMapper.class);
+
+	private static final HashMap<String, List<String>> materialTypesMap;
+
+	static
+	{
+		materialTypesMap = new HashMap<>();
```

Comment:
okay for now but longer term this is way better off somewhere in an emx file, imported and maintained outside this mapper